src/Generator/CreateTargetStatementsGenerator.php

```diff
@@ -218,6 +189,24 @@ private function constructorArgument(GeneratorMetadata $metadata, PropertyMetada
         /* Get extract and transform statements for this property */
         [$output, $propStatements] = $propertyMetadata->transformer->transform($fieldValueExpr, $constructVar, $propertyMetadata, $variableRegistry->getUniqueVariableScope(), $variableRegistry->getSourceInput());
 
+        $defaultValueExpr = new Expr\Throw_(
```

Comment:
question: why don't we throw like... now? we know the mapping process will create an error. And it has the advantage to occur sooner, moreover if the mappers generation occur at warmup time

Reply:
The mapping will not throw if you provide default constructor argument in the context, there may be case where this is needed

Reply:
ok ineed! I knew I was missing something :sweat_smile: